src/kernel-install/kernel-install.c

```diff
@@ -0,0 +1,1338 @@
+
+                free(joined);
+
+                joined = strv_join_full(strv_skip(c->argv, 1), " ", arg_root, /* escape_separator = */ false);
+                log_debug("Plugin arguments: %s", strna(joined));
+        }
+
+        r = execute_strv(
+                        /* name = */ NULL,
+                        c->plugins,
+                        arg_root,
+                        USEC_INFINITY,
+                        /* callbacks = */ NULL,
+                        /* callback_args = */ NULL,
+                        c->argv,
+                        c->envp,
+                        EXEC_DIR_SKIP_REMAINING);
+        if (r < 0)
+                return r;
+
+        r = context_remove_entry_dir(c);
+        if (r < 0)
+                return r;
+
+        return 0;
+}
+
+static int verb_add(int argc, char *argv[], void *userdata) {
+        Context *c = ASSERT_PTR(userdata);
+        int r;
+
+        assert(argc >= 3);
+        assert(argv);
+
+        c->action = ACTION_ADD;
+
+        r = context_set_version(c, argv[1]);
+        if (r < 0)
+                return r;
+
+        r = context_set_kernel(c, argv[2]);
+        if (r < 0)
+                return r;
+
+        r = context_set_initrds(c, strv_skip(argv, 3));
+        if (r < 0)
+                return r;
+
+        r = context_prepare_execution(c);
+        if (r < 0)
+                return r;
+
+        return context_execute(c);
+}
+
+static int run_as_installkernel(int argc, char *argv[], Context *c) {
+        /* kernel's install.sh invokes us as
+         *   /sbin/installkernel <version> <vmlinuz> <map> <installation-dir>
+         * We ignore the last two arguments. */
+        if (optind + 2 > argc)
+                return log_error_errno(SYNTHETIC_ERRNO(EINVAL), "'installkernel' command requires at least two arguments.");
+
+        return verb_add(3, STRV_MAKE("add", argv[optind], argv[optind+1]), c);
+}
+
+static int verb_remove(int argc, char *argv[], void *userdata) {
+        Context *c = ASSERT_PTR(userdata);
+        int r;
+
+        assert(argc == 2);
+        assert(argv);
+
+        c->action = ACTION_REMOVE;
+
+        r = context_set_version(c, argv[1]);
+        if (r < 0)
+                return r;
+
+        r = context_prepare_execution(c);
+        if (r < 0)
+                return r;
+
+        return context_execute(c);
+}
+
+static int verb_inspect(int argc, char *argv[], void *userdata) {
+        Context *c = ASSERT_PTR(userdata);
+        _cleanup_free_ char *prefix = NULL, *joined = NULL;
+        int r;
+
+        c->action = ACTION_INSPECT;
+
+        if (argc >= 2) {
+                r = context_set_kernel(c, argv[1]);
+                if (r < 0)
+                        return r;
+        }
+
+        r = context_prepare_execution(c);
+        if (r < 0)
+                return r;
+
+        prefix = strjoin("  ", strempty(arg_root));
+        if (!prefix)
+                return log_oom();
+
+        puts("Plugins:");
```

Comment:
How about
```suggestion
        puts("Using plugins:");
```
for clarity that these are the plugins that were selected for use (usually automatically)?